### ISSUE TITLE
Getting mask logits

### ISSUE THREAD

**Issue by kentaroy47** [open-mmlab/mmdetection#5133]
Thanks for the development of a great framework.

My question is : Is there a way to get raw mask logits of the inference results?
This is helpful since we can tune the mask threshold by post-processing.

Thanks!

**Comment by RangiLyu** [comment]
The default mask threshold setting is `test_cfg=dict(rcnn=dict(mask_thr_binary=0.5)))`, you can change the value in the config. If you want to get mask logits for debugging, you can set mask_thr_binary=-1 to get a uint8 mask.

**Comment by kentaroy47** [comment]
@RangiLyu 
> If you want to get mask logits for debugging, you can set mask_thr_binary=-1 to get a uint8 mask.

This will fulfil my use case, thanks for the suggestion!

**Comment by kentaroy47** [comment]
@RangiLyu 
I tried setting `model.test_cfg.rcnn.mask_thr_binary=-1` during test, but the decoded mask output was a ordinary binary mask. Am I missing something..?

**Comment by RangiLyu** [comment]
> @RangiLyu
> I tried setting `model.test_cfg.rcnn.mask_thr_binary=-1` during test, but the decoded mask output was a ordinary binary mask. Am I missing something..?
> 
> to be specific, I ran inference with
> 
> ```
> python tools/test.py $cfg $check --format-only --cfg-options model.test_cfg.rcnn.mask_thr_binary=-1
> ```
> 
> and simply decoded the output mask as
> 
> ```
> from pycocotools import mask
> mask.decode(data["segmentation"])
> ```
> 
> EDIT:
> the output mask was very strange.
> pycocotools can handle only binary masks?

pycocotools uses RLE to encode mask results to json format. It can only handle binary masks.

**Comment by kentaroy47** [comment]
@RangiLyu 
Thanks for the comments. 
mmdetection seems to use pycocotools+RLE to encode masks by default.

So I should rewrite the mask-outputting function to obtain the logit mask values?

**Comment by ZwwWayne** [comment]
Yes you can modify it here https://github.com/open-mmlab/mmdetection/blob/master/mmdet/apis/test.py#L60. We will refactor this in the future thus we can do that by modifying configs.
